twine/commands/upload.py

```diff
@@ -134,8 +134,8 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     repository.close()
 
 
-def main(args: List[str]) -> None:
-    parser = argparse.ArgumentParser(prog="twine upload")
+def main(root_parser: argparse._SubParsersAction) -> None:
+    parser = root_parser.add_parser(name="upload", description="upload a file to PyPI")
```

Comment:
I'm wary of changing Twine's public API, and using a private API of `argparse`.

Reply:
I think you're mistaken. That private class is the return value of a public API of argparse, so IMHO here that being private is wrong. Clearly any change to the return object would be a breaking change, so IMHO this is more of a side effect who wrote that class private did not annotate it correctly.